titus-server-gateway/src/main/java/com/netflix/titus/gateway/service/v3/internal/TaskDataInjector.java

```diff
@@ -223,26 +228,107 @@ private Task newTaskWithContainerState(Task task) {
         if (pods.get(taskId) == null) {
             return Collections.emptyList();
         }
-        List<ContainerStatus> containerStatuses = pods.get(taskId).getStatus().getContainerStatuses();
+        Pod pod = pods.get(taskId);
+        List<ContainerStatus> containerStatuses = pod.getStatus().getContainerStatuses();
         if (CollectionsExt.isNullOrEmpty(containerStatuses)) {
             return Collections.emptyList();
         }
         List<TaskStatus.ContainerState> containerStates = new ArrayList<>();
         for (ContainerStatus containerStatus : containerStatuses) {
             ContainerState status = containerStatus.getState();
+            String containerName = containerStatus.getName();
             TaskStatus.ContainerState.ContainerHealth containerHealth = TaskStatus.ContainerState.ContainerHealth.Unset;
             if (status.getRunning() != null) {
                 containerHealth = TaskStatus.ContainerState.ContainerHealth.Healthy;
             } else if (status.getTerminated() != null) {
                 containerHealth = TaskStatus.ContainerState.ContainerHealth.Unhealthy;
             }
+            BasicImage basicImage = buildBasicImageFromContainerStatus(pod, containerStatus.getImage(), containerName);
+            String platformSidecar = getPlatformSidecarThatCreatedContainer(pod, containerName);
             containerStates.add(TaskStatus.ContainerState.newBuilder()
                     .setContainerName(containerStatus.getName())
-                    .setContainerHealth(containerHealth).build());
+                    .setContainerHealth(containerHealth)
+                    .setContainerImage(basicImage)
+                    .setPlatformSidecar(platformSidecar)
+                    .build());
         }
         return containerStates;
     }
 
+    /**
+     * Generates a Titus object for basic image from then general string object that k8s provides
+     * (the kind you might provide to `docker pull`)
+     * But the original tag is often not available. For that we have to look at annotations to see
+     * the tag that generated the digest (if available, not all images come from tags).
+     */
+    static BasicImage buildBasicImageFromContainerStatus(Pod pod, String image, String containerName) {
+        BasicImage.Builder bi = BasicImage.newBuilder();
+        String imageWithoutRegistry = stripRegistryFromImage(image);
+        String name = getNameFromImageString(imageWithoutRegistry);
+        bi.setName(name);
+        bi.setDigest(getDigestFromImageString(imageWithoutRegistry));
+        bi.setTag(getTagFromImageString(imageWithoutRegistry, containerName, pod));
+        return bi.build();
+    }
+
+    private static String stripRegistryFromImage(String image) {
```

Comment:
All these methods might be useful in other places as well. We should have them in `KubeUtil` or something like that.